### PR TITLE
chore: signaleringen rest types

### PR DESCRIPTION
**Pull request by xiduzo**
refactor the use of "signaleringen" interfaces to use the `GeneratedType` over the custom defined ones

Solves [PZ-4799](https://dimpact.atlassian.net/browse/PZ-4799)

### CODE REVIEW

File: src/main/app/src/app/admin/groep-signaleringen/groep-signaleringen.component.ts [hunk 1]
```diff
@@ -30,8 +30,9 @@ export class GroepSignaleringenComponent
   groepen: Observable<Group[]>;
   groepId: string;
   columns: string[] = ["subjecttype", "type", "dashboard", "mail"];
-  dataSource: MatTableDataSource<SignaleringSettings> =
-    new MatTableDataSource<SignaleringSettings>();
+  dataSource = new MatTableDataSource<
+    GeneratedType<"RestSignaleringInstellingen">
```
Comment (bas-info-nl) [inline src/main/app/src/app/admin/groep-signaleringen/groep-signaleringen.component.ts:34]: is this the correct formatting? Looks weird to me 😉

File: src/main/app/src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -39,7 +38,7 @@ export class DashboardComponent implements OnInit {
     new DashboardCard(
       DashboardCardId.MIJN_TAKEN_NIEUW,
       DashboardCardType.TAKEN,
-      SignaleringType.TAAK_OP_NAAM,
+      "TAAK_OP_NAAM",
```
Comment (bas-info-nl) [inline src/main/app/src/app/dashboard/dashboard.component.ts:41]: This is a shame, using static strings where they used to be enums... Can't we at least make them constant values?

**Review by bas-info-nl — APPROVED** [review]

File: src/main/app/src/app/dashboard/dashboard.component.ts (hunk 2 above)
Comment (xiduzo) [reply src/main/app/src/app/dashboard/dashboard.component.ts:41]: Kind off, but that is how they currently come from the generated types. It is still completely type safe though, so they are half-magic-strings

image

File: src/main/app/src/app/admin/groep-signaleringen/groep-signaleringen.component.ts (hunk 1 above)
Comment (xiduzo) [reply src/main/app/src/app/admin/groep-signaleringen/groep-signaleringen.component.ts:34]: It is, spotless made it this way.

`dataSource = new MatTableDataSource<GeneratedType<"RestSignaleringInstellingen">>();` is just too long to be on the same line for formatting

File: src/main/app/src/app/dashboard/dashboard.component.ts (hunk 2 above)
Comment (xiduzo) [reply src/main/app/src/app/dashboard/dashboard.component.ts:41]: ☝️ They are type save as the `DashboardCard` does require the type to be of a specific type

```typescript
constructor(
    id: DashboardCardId,
    type: DashboardCardType,
    signaleringType?: GeneratedType<"RestSignaleringInstellingen">["type"],
  )
```

**Pull request merged**